### PR TITLE
Feature/topologicalsort2

### PR DESCRIPTION
**Pull request by niekvanderlaan**
This introduces a topological order of drawing the graph.

### CODE REVIEW

File: src/main/java/application/controllers/GraphController.java [hunk 1]
```diff
@@ -374,9 +374,7 @@ public Image takeSnapshot() {
             pref = 2500;
         }
 
-        WritableImage image;
-
-        if (graph.getModel().getAllCells().isEmpty()) {
+        WritableImage image;if (graph.getModel().getAllCells().isEmpty()) {
```
Comment (ric-jon) [inline src/main/java/application/controllers/GraphController.java]: This cannot be put on the same line

File: src/main/java/core/graph/Graph.java [hunk 2]
```diff
@@ -125,6 +123,7 @@ public Boolean addGraphComponents(ArrayList<String> ref, int depth) {
                     currentInt = depth;
                 } else if (ref != currentRef) {
                     currentRef = ref;
+                    System.out.println("!=");
```
Comment (ric-jon) [inline src/main/java/core/graph/Graph.java]: debug sout can be removed

File: src/main/java/core/graph/Graph.java [hunk 3]
```diff
@@ -269,56 +362,23 @@ private void generateModelWithSelectedGenomes(HashMap<Integer, Node> nodeMap, No
...
-    }
-
-    /**
-     * Method to give each cell to the edges
-     * @param to source
-     * @param from sink
-     * @param nodeMap the current NodeMap
-     * @param toret the new Model
-     * @param ref the currently highlighted strain
+<<<<<<< HEAD
```
Comment (ric-jon) [inline src/main/java/core/graph/Graph.java]: merge conflict leftover

File: src/main/java/application/fxobjects/layout/GraphLayout.java [hunk 4]
```diff
@@ -105,63 +121,60 @@ public void execute() {
...
+    public void breadthFirstPlacing(Cell cell, List<Cell> childrenToDraw) {
         int yOffset = 2 * offset; //y-offset between nodes on the same x-level
         int oddChildOffset = 0; //initial offset when there are an odd number of children
         int evenChildOffset = yOffset / 2; //offset for an even amount of children
         int modifier = -1; //alternate between above and below for the same x-level
         for (Cell c : cell.getCellChildren()) {
             GraphCell child = (GraphCell) c;
-            if (!child.isRelocated() || child.getLayoutX() < cell.getLayoutX()) {
                 if (cellCount % 2 == 0) {
+                    double yCoordinate = currentY - evenChildOffset
```
Comment (Warnars) [inline src/main/java/application/fxobjects/layout/GraphLayout.java]: Can this be placed on one line

**Comment by tieswesterborg** [comment]
No further remarks.

**Pull request merged**